### ISSUE TITLE
Migrate to eslint@2.0.0

### ISSUE THREAD

**Issue by deckar01** [tmpvar/jsdom#1446]
When https://github.com/lo1tuma/eslint-plugin-mocha/issues/57 is fixed it will depend on `eslint@2.0.0` unless they release a patch for `1.1`.

My initial test indicated that simply resolving the deprecated rules did not cause eslint to pass.

**Comment by domenic** [comment]
I gave this a try. Blocked on https://github.com/eslint/eslint/pull/5661.

**Comment by deckar01** [comment]
There seems to be quite a few places where the config file had a rule set to "error", but there were violations getting missed before 2.0.0.

**Comment by domenic** [comment]
Yeah, probably the whole-program analysis thing they introduced. I started in the https://github.com/tmpvar/jsdom/tree/eslint-2 branch but didn't finish, and probably won't bother until they fix the quotes rule anyway.

**Comment by deckar01** [comment]
It would be nice if `--fix` was implemented in more of the core rules.

**Comment by deckar01** [comment]
I had a feeling this was going to cause a lot of changes.

When I originally looked at this, the html file [script-languages-02.html](https://github.com/tmpvar/jsdom/blob/93dd663eba2e86bd7a6562edeccefa4ebe3da2ac/test/web-platform-tests/to-upstream/html/semantics/scripting-1/the-script-element/script-languages-02.html) gave me trouble, because it was es5. I tried configuring an eslint hint to handle it correctly, but it did not seem to work. I noticed you set the jslint hint to ignore that script, which seems reasonable for test data, just curious what your thoughts were.
